### ISSUE TITLE
gdb 7.12.1 fails to compile with 1.23

### ISSUE THREAD

**Issue by cleeland** [crosstool-ng/crosstool-ng#901]
### Issue submission guidelines
See [detaled guidelines](http://crosstool-ng.github.io/support/). In short:
- Report crosstool-NG version
This is crosstool-NG version crosstool-ng-1.23.0

- Report host OS and its version
MacOS Sierra 10.12.6
homebrew

- Attach `.config` (unless the issue is not related to any configuration)
- Attach `build.log` (unless crosstool-NG fails before the build starts)
[bugreport.zip](https://github.com/crosstool-ng/crosstool-ng/files/1654643/bugreport.zip)

There is a bug in gdb related to use of `min` and `max` such that 7.12.1 fails to compile under certain circumstances...such as on MacOS or with clang and libc++.  There is a bug report for the same problem on freebsd at https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=219126

I found that applying the first of the two listed upstream patches enabled gdb to compile just fine, and I included that patch in the attached zip file.

**Comment by cpackham** [comment]
Hi @cleeland,

I take it your build.log is from a run with the local patch applied? In other words the build.log suggests compilation was successful

**Comment by cleeland** [comment]
Yes it was.  I hand-patched (using the same `patch` command line args that `CT_Patch` uses) and restarted the build.

I suppose I should rebuild the whole thing again to ensure it actually works with the automated, but I was so close to completing the toolchain that I just wanted to get to the end.
